lib/nodes/Parent.ts

```diff
@@ -0,0 +1,76 @@
+import {Base, BaseNode} from './Base';
+import {BlueshellState} from './BlueshellState';
+
+/**
+ * Interface for a node that exposes a list of children.
+ * Note: because you can't have an abstract getter/property, getChildren is just a normal method.
+ */
+export interface ParentNode<S extends BlueshellState, E> extends BaseNode<S, E> {
+	getChildren(): Base<S, E>[];
+}
+
+/**
+ * Checks if the passed in Node exposes a list of children.
+ * @param node Node to check
+ */
+export function isParentNode<S extends BlueshellState, E>(
+	node: BaseNode<S, E>
```

Comment:
for a type guard, I recommend making the input `any` so you don't have to cast. Typeguard arguments in my opinion are one of the only times where `any` is the right type for something.

Reply:
I made that change (the function moved files to ParentNode.ts)